Added dependencies to Dockerfile required for running CasparCG inside container

saltomodules opened the pull request:
The final docker image build by the 'build in docker' process provided from the main CasparCG repo misses dependencies needed by CasparCG at runtime. 

Trying to run CasparCG in a container spawned from this image therefore doesn't work at the moment.

This pull request adds the required lib installs to the main Dockerfile to remedy that.

Julusian commented:
Strange, line 36 `/source/shell/copy_deps.sh shell/casparcg /staging/lib` is supposed to be copying the so files needed. Perhaps that line is missing something that is needed?

It used to work fine without having to install that list you added, but that was probably before I changed the version of ubuntu it was using

saltomodules commented:
I looked a little further into this.

The `copy_deps.sh` script does seem to work. A long list of `.so` files is present in the final docker image in `/opt/casparcg/lib`.

With `ldd /opt/casparcg/bin/casparcg` you can see that some libs are actually found in the `./../lib` folder relative to the casparcg executable. However a lot of the libs are not found, even though they are present in that lib folder:


<details>
  <summary>.so files present in /opt/casparcg/lib</summary>
  
  ```root@845253e56310:/opt/casparcg/lib# ll
total 370328
drwxr-xr-x 4 root root     12288 Sep 12 00:03 ./
drwxr-xr-x 1 root root      4096 Sep 12 00:04 ../
-rwxr-xr-x 1 root root   1624936 Sep 12 00:02 chrome-sandbox*
-rw-r--r-- 1 root root    630881 Sep 12 00:02 chrome_100_percent.pak
-rw-r--r-- 1 root root    944015 Sep 12 00:02 chrome_200_percent.pak
-rw-r--r-- 1 root root  10215872 Sep 12 00:02 icudtl.dat
-rwxr-xr-x 1 root root   2980848 Sep 12 00:02 libEGL.so*
-rw-r--r-- 1 root root    247896 Sep 12 00:03 libFLAC.so.8
-rw-r--r-- 1 root root    547152 Sep 12 00:03 libGL.so.1
-rwxr-xr-x 1 root root  30945888 Sep 12 00:02 libGLESv2.so*
-rw-r--r-- 1 root root    669584 Sep 12 00:03 libGLEW.so.2.1
-rw-r--r-- 1 root root    141912 Sep 12 00:03 libGLX.so.0
-rw-r--r-- 1 root root    715200 Sep 12 00:03 libGLdispatch.so.0
-rw-r--r-- 1 root root    276776 Sep 12 00:03 libHalf.so.24
-rw-r--r-- 1 root root    129608 Sep 12 00:03 libIex-2_3.so.24
-rw-r--r-- 1 root root   3000752 Sep 12 00:03 libIlmImf-2_3.so.24
-rw-r--r-- 1 root root     35512 Sep 12 00:03 libIlmThread-2_3.so.24
-rw-r--r-- 1 root root   1368744 Sep 12 00:03 libSDL2-2.0.so.0
-rw-r--r-- 1 root root   1293928 Sep 12 00:03 libX11.so.6
-rw-r--r-- 1 root root     18688 Sep 12 00:03 libXau.so.6
-rw-r--r-- 1 root root     14488 Sep 12 00:03 libXcomposite.so.1
-rw-r--r-- 1 root root     47584 Sep 12 00:03 libXcursor.so.1
-rw-r--r-- 1 root root     14512 Sep 12 00:03 libXdamage.so.1
-rw-r--r-- 1 root root     26800 Sep 12 00:03 libXdmcp.so.6
-rw-r--r-- 1 root root     81600 Sep 12 00:03 libXext.so.6
-rw-r--r-- 1 root root     26824 Sep 12 00:03 libXfixes.so.3
-rw-r--r-- 1 root root     68128 Sep 12 00:03 libXi.so.6
-rw-r--r-- 1 root root     14496 Sep 12 00:03 libXinerama.so.1
-rw-r--r-- 1 root root     47504 Sep 12 00:03 libXrandr.so.2
-rw-r--r-- 1 root root     39448 Sep 12 00:03 libXrender.so.1
-rw-r--r-- 1 root root     14528 Sep 12 00:03 libXss.so.1
-rw-r--r-- 1 root root     18800 Sep 12 00:03 libXv.so.1
-rw-r--r-- 1 root root     22872 Sep 12 00:03 libXxf86vm.so.1
-rw-r--r-- 1 root root     80736 Sep 12 00:03 libapparmor.so.1
-rw-r--r-- 1 root root   1022400 Sep 12 00:03 libasound.so.2
-rw-r--r-- 1 root root    207080 Sep 12 00:03 libass.so.9
-rw-r--r-- 1 root root     22368 Sep 12 00:03 libasyncns.so.0
-rw-r--r-- 1 root root    165968 Sep 12 00:03 libatk-1.0.so.0
-rw-r--r-- 1 root root    219008 Sep 12 00:03 libatk-bridge-2.0.so.0
-rw-r--r-- 1 root root    219464 Sep 12 00:03 libatspi.so.0
-rw-r--r-- 1 root root     72728 Sep 12 00:03 libavahi-client.so.3
-rw-r--r-- 1 root root     52136 Sep 12 00:03 libavahi-common.so.3
-rw-r--r-- 1 root root    351352 Sep 12 00:03 libblkid.so.1
-rw-r--r-- 1 root root    326528 Sep 12 00:03 libbluray.so.2
-rw-r--r-- 1 root root     96728 Sep 12 00:03 libbsd.so.0
-rw-r--r-- 1 root root     74848 Sep 12 00:03 libbz2.so.1.0
-rw-r--r-- 1 root root   1183496 Sep 12 00:03 libcairo.so.2
-rwxr-xr-x 1 root root 187712736 Sep 12 00:03 libcef.so*
-rw-r--r-- 1 root root     22600 Sep 12 00:03 libcom_err.so.2
-rw-r--r-- 1 root root    628848 Sep 12 00:03 libcups.so.2
-rw-r--r-- 1 root root     34888 Sep 12 00:03 libdatrie.so.1
-rw-r--r-- 1 root root    325944 Sep 12 00:03 libdbus-1.so.3
-rw-r--r-- 1 root root     18816 Sep 12 00:03 libdl.so.2
-rw-r--r-- 1 root root     84600 Sep 12 00:03 libdrm.so.2
-rw-r--r-- 1 root root    182560 Sep 12 00:03 libexpat.so.1
-rw-r--r-- 1 root root     43416 Sep 12 00:03 libffi.so.7
-rw-r--r-- 1 root root    285840 Sep 12 00:03 libfontconfig.so.1
-rw-r--r-- 1 root root    765336 Sep 12 00:03 libfreeimage.so.3
-rw-r--r-- 1 root root    776264 Sep 12 00:03 libfreetype.so.6
-rw-r--r-- 1 root root    112960 Sep 12 00:03 libfribidi.so.0
-rw-r--r-- 1 root root     64624 Sep 12 00:03 libgbm.so.1
-rw-r--r-- 1 root root    104984 Sep 12 00:03 libgcc_s.so.1
-rw-r--r-- 1 root root   1168056 Sep 12 00:03 libgcrypt.so.20
-rw-r--r-- 1 root root   1957200 Sep 12 00:03 libgio-2.0.so.0
-rw-r--r-- 1 root root   1207920 Sep 12 00:03 libglib-2.0.so.0
-rw-r--r-- 1 root root     18640 Sep 12 00:03 libgmodule-2.0.so.0
-rw-r--r-- 1 root root    534880 Sep 12 00:03 libgmp.so.10
-rw-r--r-- 1 root root   1914336 Sep 12 00:03 libgnutls.so.30
-rw-r--r-- 1 root root    387448 Sep 12 00:03 libgobject-2.0.so.0
-rw-r--r-- 1 root root    265856 Sep 12 00:03 libgomp.so.1
-rw-r--r-- 1 root root    137584 Sep 12 00:03 libgpg-error.so.0
-rw-r--r-- 1 root root    178488 Sep 12 00:03 libgraphite2.so.3
-rw-r--r-- 1 root root    309712 Sep 12 00:03 libgssapi_krb5.so.2
-rw-r--r-- 1 root root   1065824 Sep 12 00:03 libharfbuzz.so.0
-rw-r--r-- 1 root root    219976 Sep 12 00:03 libhogweed.so.5
-rw-r--r-- 1 root root  28046896 Sep 12 00:03 libicudata.so.66
-rw-r--r-- 1 root root   3132040 Sep 12 00:03 libicui18n.so.66
-rw-r--r-- 1 root root   1976648 Sep 12 00:03 libicuuc.so.66
-rw-r--r-- 1 root root    129096 Sep 12 00:03 libidn2.so.0
-rw-r--r-- 1 root root     58216 Sep 12 00:03 libjbig.so.0
-rw-r--r-- 1 root root    539352 Sep 12 00:03 libjpeg.so.8
-rw-r--r-- 1 root root    211264 Sep 12 00:03 libjpegxr.so.0
-rw-r--r-- 1 root root    127448 Sep 12 00:03 libjxrglue.so.0
-rw-r--r-- 1 root root    191040 Sep 12 00:03 libk5crypto.so.3
-rw-r--r-- 1 root root     22600 Sep 12 00:03 libkeyutils.so.1
-rw-r--r-- 1 root root    902016 Sep 12 00:03 libkrb5.so.3
-rw-r--r-- 1 root root     56096 Sep 12 00:03 libkrb5support.so.0
-rw-r--r-- 1 root root    365568 Sep 12 00:03 liblcms2.so.2
-rw-r--r-- 1 root root    129248 Sep 12 00:03 liblz4.so.1
-rw-r--r-- 1 root root    162264 Sep 12 00:03 liblzma.so.5
-rw-r--r-- 1 root root   1369352 Sep 12 00:03 libm.so.6
-rw-r--r-- 1 root root    387768 Sep 12 00:03 libmount.so.1
-rw-r--r-- 1 root root    301192 Sep 12 00:03 libmp3lame.so.0
-rw-r--r-- 1 root root    323320 Sep 12 00:03 libmpg123.so.0
-rw-r--r-- 1 root root    231592 Sep 12 00:03 libnettle.so.7
-rw-r--r-- 1 root root    105528 Sep 12 00:03 libnsl.so.1
-rw-r--r-- 1 root root    247664 Sep 12 00:03 libnspr4.so
-rw-r--r-- 1 root root   1363376 Sep 12 00:03 libnss3.so
-rw-r--r-- 1 root root    573112 Sep 12 00:03 libnssckbi.so
-rw-r--r-- 1 root root    203808 Sep 12 00:03 libnssutil3.so
-rw-r--r-- 1 root root     47960 Sep 12 00:03 libnuma.so.1
-rw-r--r-- 1 root root     47064 Sep 12 00:03 libogg.so.0
-rw-r--r-- 1 root root    925176 Sep 12 00:03 libopenal.so.1
-rw-r--r-- 1 root root    170200 Sep 12 00:03 libopencore-amrnb.so.0
-rw-r--r-- 1 root root     84104 Sep 12 00:03 libopencore-amrwb.so.0
-rw-r--r-- 1 root root    346592 Sep 12 00:03 libopenjp2.so.7
-rw-r--r-- 1 root root   1677680 Sep 12 00:03 libopenmpt.so.0
-rw-r--r-- 1 root root    383048 Sep 12 00:03 libopus.so.0
-rw-r--r-- 1 root root   1265624 Sep 12 00:03 libp11-kit.so.0
-rw-r--r-- 1 root root    317512 Sep 12 00:03 libpango-1.0.so.0
-rw-r--r-- 1 root root    465008 Sep 12 00:03 libpcre.so.3
-rw-r--r-- 1 root root    584392 Sep 12 00:03 libpcre2-8.so.0
-rw-r--r-- 1 root root    678064 Sep 12 00:03 libpixman-1.so.0
-rw-r--r-- 1 root root     22928 Sep 12 00:03 libplc4.so
-rw-r--r-- 1 root root     14640 Sep 12 00:03 libplds4.so
-rw-r--r-- 1 root root    223304 Sep 12 00:03 libpng16.so.16
-rwxr-xr-x 1 root root    157224 Sep 12 00:03 libpthread.so.0*
-rw-r--r-- 1 root root    343240 Sep 12 00:03 libpulse.so.0
-rw-r--r-- 1 root root    526688 Sep 12 00:03 libpulsecommon-13.99.so
-rw-r--r-- 1 root root    911496 Sep 12 00:03 libraw.so.19
-rw-r--r-- 1 root root    101320 Sep 12 00:03 libresolv.so.2
-rw-r--r-- 1 root root     35928 Sep 12 00:03 librt.so.1
-rw-r--r-- 1 root root    163200 Sep 12 00:03 libselinux.so.1
-rw-r--r-- 1 root root    355088 Sep 12 00:03 libsfml-graphics.so.2.5
-rw-r--r-- 1 root root     51360 Sep 12 00:03 libsfml-system.so.2.5
-rw-r--r-- 1 root root    162184 Sep 12 00:03 libsfml-window.so.2.5
-rw-r--r-- 1 root root     43104 Sep 12 00:03 libshine.so.3
-rw-r--r-- 1 root root    193032 Sep 12 00:03 libsmime3.so
-rw-r--r-- 1 root root     39272 Sep 12 00:03 libsnappy.so.1
-rw-r--r-- 1 root root    504048 Sep 12 00:03 libsndfile.so.1
-rw-r--r-- 1 root root     55520 Sep 12 00:03 libsndio.so.7.0
-rw-r--r-- 1 root root    311568 Sep 12 00:03 libsoftokn3.so
-rw-r--r-- 1 root root    216120 Sep 12 00:03 libsoxr.so.0
-rw-r--r-- 1 root root    116776 Sep 12 00:03 libspeex.so.1
-rw-r--r-- 1 root root    709496 Sep 12 00:03 libsystemd.so.0
-rw-r--r-- 1 root root     84120 Sep 12 00:03 libtasn1.so.6
-rw-r--r-- 1 root root    272840 Sep 12 00:03 libtbb.so.2
-rw-r--r-- 1 root root    132976 Sep 12 00:03 libtbbmalloc.so.2
-rw-r--r-- 1 root root     14576 Sep 12 00:03 libtbbmalloc_proxy.so.2
-rw-r--r-- 1 root root     41152 Sep 12 00:03 libthai.so.0
-rw-r--r-- 1 root root    125232 Sep 12 00:03 libtheoradec.so.1
-rw-r--r-- 1 root root    244016 Sep 12 00:03 libtheoraenc.so.1
-rw-r--r-- 1 root root    522936 Sep 12 00:03 libtiff.so.5
-rw-r--r-- 1 root root    133192 Sep 12 00:03 libtwolame.so.0
-rw-r--r-- 1 root root    178528 Sep 12 00:03 libudev.so.1
-rw-r--r-- 1 root root   1575112 Sep 12 00:03 libunistring.so.2
-rw-r--r-- 1 root root     30936 Sep 12 00:03 libuuid.so.1
-rwxr-xr-x 1 root root  26147168 Sep 12 00:02 libvk_swiftshader.so*
-rw-r--r-- 1 root root    111440 Sep 12 00:03 libvo-amrwbenc.so.0
-rw-r--r-- 1 root root    182344 Sep 12 00:03 libvorbis.so.0
-rw-r--r-- 1 root root    694408 Sep 12 00:03 libvorbisenc.so.2
-rw-r--r-- 1 root root     38904 Sep 12 00:03 libvorbisfile.so.3
-rw-r--r-- 1 root root   2308168 Sep 12 00:03 libvpx.so.6
-rwxr-xr-x 1 root root   2334264 Sep 12 00:02 libvulkan.so.1*
-rw-r--r-- 1 root root    174216 Sep 12 00:03 libwavpack.so.1
-rw-r--r-- 1 root root     64808 Sep 12 00:03 libwayland-client.so.0
-rw-r--r-- 1 root root     39448 Sep 12 00:03 libwayland-cursor.so.0
-rw-r--r-- 1 root root     14368 Sep 12 00:03 libwayland-egl.so.1
-rw-r--r-- 1 root root     85448 Sep 12 00:03 libwayland-server.so.0
-rw-r--r-- 1 root root    420376 Sep 12 00:03 libwebp.so.6
-rw-r--r-- 1 root root     43336 Sep 12 00:03 libwebpmux.so.3
-rw-r--r-- 1 root root     44848 Sep 12 00:03 libwrap.so.0
-rw-r--r-- 1 root root   1851544 Sep 12 00:03 libx264.so.155
-rw-r--r-- 1 root root  16108128 Sep 12 00:03 libx265.so.179
-rw-r--r-- 1 root root     55544 Sep 12 00:03 libxcb-render.so.0
-rw-r--r-- 1 root root     14568 Sep 12 00:03 libxcb-shm.so.0
-rw-r--r-- 1 root root    166488 Sep 12 00:03 libxcb.so.1
-rw-r--r-- 1 root root    264768 Sep 12 00:03 libxkbcommon.so.0
-rw-r--r-- 1 root root   1799408 Sep 12 00:03 libxml2.so.2
-rw-r--r-- 1 root root      6160 Sep 12 00:03 libxshmfence.so.1
-rw-r--r-- 1 root root    691200 Sep 12 00:03 libxvidcore.so.4
-rw-r--r-- 1 root root    108936 Sep 12 00:03 libz.so.1
-rw-r--r-- 1 root root    686160 Sep 12 00:03 libzstd.so.1
drwxr-xr-x 2 root root      4096 Sep 12 00:02 locales/
-rw-r--r-- 1 root root   6589757 Sep 12 00:02 resources.pak
-rw-r--r-- 1 root root     49443 Sep 12 00:02 snapshot_blob.bin
drwxr-xr-x 2 root root      4096 Sep 12 00:02 swiftshader/
-rw-r--r-- 1 root root    164687 Sep 12 00:02 v8_context_snapshot.bin
-rw-r--r-- 1 root root       107 Sep 12 00:02 vk_swiftshader_icd.json
```
</details>

<details>
  <summary>ldd output</summary>
  
  ```root@845253e56310:/opt/casparcg/bin# ldd casparcg
	linux-vdso.so.1 (0x00007ffd2e4ec000)
	libtbb.so.2 => /opt/casparcg/bin/./../lib/libtbb.so.2 (0x00007fb5e61ae000)
	libtbbmalloc.so.2 => /opt/casparcg/bin/./../lib/libtbbmalloc.so.2 (0x00007fb5e616c000)
	libtbbmalloc_proxy.so.2 => /opt/casparcg/bin/./../lib/libtbbmalloc_proxy.so.2 (0x00007fb5e6167000)
	libsfml-graphics.so.2.5 => /opt/casparcg/bin/./../lib/libsfml-graphics.so.2.5 (0x00007fb5e610f000)
	libsfml-window.so.2.5 => /opt/casparcg/bin/./../lib/libsfml-window.so.2.5 (0x00007fb5e60e6000)
	libsfml-system.so.2.5 => /opt/casparcg/bin/./../lib/libsfml-system.so.2.5 (0x00007fb5e60d6000)
	libGLEW.so.2.1 => /opt/casparcg/bin/./../lib/libGLEW.so.2.1 (0x00007fb5e602c000)
	libGL.so.1 => /usr/lib/x86_64-linux-gnu/libGL.so.1 (0x00007fb5e5fa4000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007fb5e5e67000)
	libvpx.so.6 => /opt/casparcg/bin/./../lib/libvpx.so.6 (0x00007fb5e5c2f000)
	libm.so.6 => /usr/lib/x86_64-linux-gnu/libm.so.6 (0x00007fb5e5ae0000)
	libwebpmux.so.3 => /opt/casparcg/bin/./../lib/libwebpmux.so.3 (0x00007fb5e5ad2000)
	libwebp.so.6 => /opt/casparcg/bin/./../lib/libwebp.so.6 (0x00007fb5e5a68000)
	liblzma.so.5 => /usr/lib/x86_64-linux-gnu/liblzma.so.5 (0x00007fb5e5a3f000)
	libopencore-amrwb.so.0 => /opt/casparcg/bin/./../lib/libopencore-amrwb.so.0 (0x00007fb5e5a29000)
	libsnappy.so.1 => /opt/casparcg/bin/./../lib/libsnappy.so.1 (0x00007fb5e5a1e000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fb5e583c000)
	libz.so.1 => /usr/lib/x86_64-linux-gnu/libz.so.1 (0x00007fb5e581e000)
	libmp3lame.so.0 => /opt/casparcg/bin/./../lib/libmp3lame.so.0 (0x00007fb5e57a6000)
	libopencore-amrnb.so.0 => /opt/casparcg/bin/./../lib/libopencore-amrnb.so.0 (0x00007fb5e577b000)
	libopenjp2.so.7 => /opt/casparcg/bin/./../lib/libopenjp2.so.7 (0x00007fb5e5725000)
	libopus.so.0 => /opt/casparcg/bin/./../lib/libopus.so.0 (0x00007fb5e56c6000)
	libshine.so.3 => /opt/casparcg/bin/./../lib/libshine.so.3 (0x00007fb5e54bb000)
	libspeex.so.1 => /opt/casparcg/bin/./../lib/libspeex.so.1 (0x00007fb5e549b000)
	libtheoraenc.so.1 => /opt/casparcg/bin/./../lib/libtheoraenc.so.1 (0x00007fb5e545e000)
	libtheoradec.so.1 => /opt/casparcg/bin/./../lib/libtheoradec.so.1 (0x00007fb5e543e000)
	libtwolame.so.0 => /opt/casparcg/bin/./../lib/libtwolame.so.0 (0x00007fb5e5418000)
	libvo-amrwbenc.so.0 => /opt/casparcg/bin/./../lib/libvo-amrwbenc.so.0 (0x00007fb5e53fc000)
	libvorbis.so.0 => /opt/casparcg/bin/./../lib/libvorbis.so.0 (0x00007fb5e53ce000)
	libvorbisenc.so.2 => /opt/casparcg/bin/./../lib/libvorbisenc.so.2 (0x00007fb5e5321000)
	libwavpack.so.1 => /opt/casparcg/bin/./../lib/libwavpack.so.1 (0x00007fb5e52f5000)
	libx264.so.155 => /opt/casparcg/bin/./../lib/libx264.so.155 (0x00007fb5e5037000)
	libx265.so.179 => /opt/casparcg/bin/./../lib/libx265.so.179 (0x00007fb5e40c9000)
	libxvidcore.so.4 => /opt/casparcg/bin/./../lib/libxvidcore.so.4 (0x00007fb5e3fb6000)
	libdl.so.2 => /usr/lib/x86_64-linux-gnu/libdl.so.2 (0x00007fb5e3fb0000)
	libsoxr.so.0 => /opt/casparcg/bin/./../lib/libsoxr.so.0 (0x00007fb5e3f43000)
	libxml2.so.2 => /opt/casparcg/bin/./../lib/libxml2.so.2 (0x00007fb5e3d89000)
	libbz2.so.1.0 => /usr/lib/x86_64-linux-gnu/libbz2.so.1.0 (0x00007fb5e3d76000)
	libopenmpt.so.0 => /opt/casparcg/bin/./../lib/libopenmpt.so.0 (0x00007fb5e3b89000)
	libbluray.so.2 => /opt/casparcg/bin/./../lib/libbluray.so.2 (0x00007fb5e3b38000)
	libgmp.so.10 => /usr/lib/x86_64-linux-gnu/libgmp.so.10 (0x00007fb5e3ab4000)
	libgnutls.so.30 => /usr/lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007fb5e38dc000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fb5e38b2000)
	libxcb-shm.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0 (0x00007fb5e38ad000)
	libasound.so.2 => /opt/casparcg/bin/./../lib/libasound.so.2 (0x00007fb5e37b2000)
	libSDL2-2.0.so.0 => /opt/casparcg/bin/./../lib/libSDL2-2.0.so.0 (0x00007fb5e365d000)
	libsndio.so.7.0 => /opt/casparcg/bin/./../lib/libsndio.so.7.0 (0x00007fb5e364c000)
	libXv.so.1 => /opt/casparcg/bin/./../lib/libXv.so.1 (0x00007fb5e3445000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007fb5e3430000)
	libass.so.9 => /opt/casparcg/bin/./../lib/libass.so.9 (0x00007fb5e31fd000)
	libfontconfig.so.1 => /usr/lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007fb5e31b6000)
	libfreetype.so.6 => /usr/lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007fb5e30f7000)
	libicui18n.so.66 => /opt/casparcg/bin/./../lib/libicui18n.so.66 (0x00007fb5e2df8000)
	libicuuc.so.66 => /opt/casparcg/bin/./../lib/libicuuc.so.66 (0x00007fb5e2c10000)
	libpthread.so.0 => /usr/lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fb5e2bed000)
	libfreeimage.so.3 => /opt/casparcg/bin/./../lib/libfreeimage.so.3 (0x00007fb5e2b31000)
	libopenal.so.1 => /opt/casparcg/bin/./../lib/libopenal.so.1 (0x00007fb5e2a49000)
	libcef.so => /opt/casparcg/bin/./../lib/libcef.so (0x00007fb5d74d0000)
	libgcc_s.so.1 => /usr/lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fb5d74b5000)
	libc.so.6 => /usr/lib/x86_64-linux-gnu/libc.so.6 (0x00007fb5d72c1000)
	libXrandr.so.2 => not found
	libudev.so.1 => /usr/lib/x86_64-linux-gnu/libudev.so.1 (0x00007fb5d728f000)
	libGLdispatch.so.0 => /usr/lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007fb5d71d7000)
	libGLX.so.0 => /usr/lib/x86_64-linux-gnu/libGLX.so.0 (0x00007fb5d71a3000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fb5e8425000)
	libogg.so.0 => not found
	libcairo.so.2 => not found
	libogg.so.0 => not found
	libnuma.so.1 => not found
	libgomp.so.1 => not found
	libmpg123.so.0 => not found
	libvorbisfile.so.3 => not found
	libp11-kit.so.0 => /usr/lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007fb5d7069000)
	libidn2.so.0 => /usr/lib/x86_64-linux-gnu/libidn2.so.0 (0x00007fb5d7048000)
	libunistring.so.2 => /usr/lib/x86_64-linux-gnu/libunistring.so.2 (0x00007fb5d6ec6000)
	libtasn1.so.6 => /usr/lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007fb5d6eae000)
	libnettle.so.7 => /usr/lib/x86_64-linux-gnu/libnettle.so.7 (0x00007fb5d6e74000)
	libhogweed.so.5 => /usr/lib/x86_64-linux-gnu/libhogweed.so.5 (0x00007fb5d6e3d000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007fb5d6e37000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007fb5d6e2f000)
	libpulse.so.0 => not found
	libXcursor.so.1 => not found
	libXinerama.so.1 => not found
	libXi.so.6 => not found
	libXrandr.so.2 => not found
	libXss.so.1 => not found
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007fb5d6e24000)
	libwayland-egl.so.1 => not found
	libwayland-client.so.0 => /usr/lib/x86_64-linux-gnu/libwayland-client.so.0 (0x00007fb5d6e13000)
	libwayland-cursor.so.0 => not found
	libxkbcommon.so.0 => not found
	librt.so.1 => /usr/lib/x86_64-linux-gnu/librt.so.1 (0x00007fb5d6e07000)
	libbsd.so.0 => /usr/lib/x86_64-linux-gnu/libbsd.so.0 (0x00007fb5d6ded000)
	libfribidi.so.0 => not found
	libharfbuzz.so.0 => not found
	libexpat.so.1 => /usr/lib/x86_64-linux-gnu/libexpat.so.1 (0x00007fb5d6dbf000)
	libuuid.so.1 => /usr/lib/x86_64-linux-gnu/libuuid.so.1 (0x00007fb5d6db4000)
	libpng16.so.16 => /usr/lib/x86_64-linux-gnu/libpng16.so.16 (0x00007fb5d6d7c000)
	libicudata.so.66 => not found
	libjxrglue.so.0 => not found
	libjpeg.so.8 => not found
	libraw.so.19 => not found
	libtiff.so.5 => not found
	libIlmImf-2_3.so.24 => not found
	libHalf.so.24 => not found
	libIex-2_3.so.24 => not found
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007fb5d6d18000)
	libglib-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fb5d6bef000)
	libnss3.so => /usr/lib/x86_64-linux-gnu/libnss3.so (0x00007fb5d6aa0000)
	libnssutil3.so => /usr/lib/x86_64-linux-gnu/libnssutil3.so (0x00007fb5d6a6d000)
	libsmime3.so => /usr/lib/x86_64-linux-gnu/libsmime3.so (0x00007fb5d6a3d000)
	libnspr4.so => /usr/lib/x86_64-linux-gnu/libnspr4.so (0x00007fb5d69fd000)
	libatk-1.0.so.0 => not found
	libatk-bridge-2.0.so.0 => not found
	libcups.so.2 => not found
	libdrm.so.2 => /usr/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007fb5d69e5000)
	libgio-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007fb5d6804000)
	libXcomposite.so.1 => not found
	libXdamage.so.1 => not found
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007fb5d67fa000)
	libXrandr.so.2 => not found
	libxkbcommon.so.0 => not found
	libgbm.so.1 => /usr/lib/x86_64-linux-gnu/libgbm.so.1 (0x00007fb5d67e9000)
	libdbus-1.so.3 => not found
	libpango-1.0.so.0 => not found
	libcairo.so.2 => not found
	libatspi.so.0 => not found
	libxshmfence.so.1 => /usr/lib/x86_64-linux-gnu/libxshmfence.so.1 (0x00007fb5d65e5000)
	libffi.so.7 => /usr/lib/x86_64-linux-gnu/libffi.so.7 (0x00007fb5d65d9000)
	libpcre.so.3 => /usr/lib/x86_64-linux-gnu/libpcre.so.3 (0x00007fb5d6564000)
	libplc4.so => /usr/lib/x86_64-linux-gnu/libplc4.so (0x00007fb5d655d000)
	libplds4.so => /usr/lib/x86_64-linux-gnu/libplds4.so (0x00007fb5d6558000)
	libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007fb5d6552000)
	libmount.so.1 => /usr/lib/x86_64-linux-gnu/libmount.so.1 (0x00007fb5d64f2000)
	libselinux.so.1 => /usr/lib/x86_64-linux-gnu/libselinux.so.1 (0x00007fb5d64c5000)
	libresolv.so.2 => /usr/lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fb5d64a9000)
	libwayland-server.so.0 => /usr/lib/x86_64-linux-gnu/libwayland-server.so.0 (0x00007fb5d6493000)
	libblkid.so.1 => /usr/lib/x86_64-linux-gnu/libblkid.so.1 (0x00007fb5d643c000)
	libpcre2-8.so.0 => /usr/lib/x86_64-linux-gnu/libpcre2-8.so.0 (0x00007fb5d63ac000)
```
</details>

I think this might have to do with commit e0bbb96 that removes `LD_LIBRARY_PATH=lib ` from `run.sh` and instead sets an `RPATH` for dependencies during build time.

 It looks like this RPATH flag is not propagating to dependencies of dependencies. For instance, toplevel dependency `libSDL2-2.0.so.0` is found, but its dependencies in turn are not, unless they are present in a system folder:
 
 ```
 from 'lddtree casparcg':
 
 libSDL2-2.0.so.0 => ./../lib/libSDL2-2.0.so.0
    libpulse.so.0 => not found
    libXcursor.so.1 => not found
    libXinerama.so.1 => not found
    libXi.so.6 => not found
    libXss.so.1 => not found
    libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1
    libwayland-egl.so.1 => not found
    libwayland-client.so.0 => /usr/lib/x86_64-linux-gnu/libwayland-client.so.0
    libwayland-cursor.so.0 => not found
    libxkbcommon.so.0 => not found
```

Patching all libs in `/opt/casparcg/lib` with `patchelf --set-rpath '$ORIGIN' libname`, to make their internal `RPATH` point to the same folder, seems to work, but feels a bit hacky.

This issue probably didn't show up earlier because full installs of Ubuntu have most needed libs present in their system folder and, hence, they are still found when starting CasparCG. Ofcourse, in that case you might be running CasparCG against a lib version that is different from the one you have built against.

What do you think? Should `LD_LIBRARY_PATH` be restored? 

Or perhaps there are ways to make `INSTALL_RPATH` propagate to subdependencies? 

Or might there be a completely different reason why not all libs are picked up from the `/opt/casparcg/libs` folder?

oscarwcl commented:
I think bringing back `LD_LIBRARY_PATH` makes sense as a simple fix.

Having said that, I do think that `copy_deps.sh` is not very reliable in general, and is very likely to break if you're shipping all the dependencies to a system with different library versions than where it was built (All the "Common Issues on Linux" listed in the readme likely come from the way dependencies are bundled). If all we want to support is running inside the docker image then it probably works well enough, but beyond that we'd be likely be better off either building a .deb package that pulls in the required system dependencies, or using a more robust bundle format like AppImage.

saltomodules commented:
Even when running the container with the `LD_LIBRARY_PATH` set there are still a few dependencies 'not found' when running the final image in a container:

```	
libatk-1.0.so.0 => not found
libatk-bridge-2.0.so.0 => not found
libcups.so.2 => not found
libXcomposite.so.1 => not found
libXdamage.so.1 => not found
libpango-1.0.so.0 => not found
libatspi.so.0 => not found
```

Turns out these are subdependencies of `libcef.so`

When running `copy_deps.sh` on line 36 `ldd` is checking `/build/shell/casparcg` for dependencies.

But `libcef.so` is `not found` from this location. Probably because `ldd` is looking for the lib in the relative path `./../lib`, which is non-existent relative to `/build/shell/casparcg`:

```
root@b2eebedc6012:/build/staging# ldd /build/shell/casparcg 
	linux-vdso.so.1 (0x00007ffda935d000)
	libtbb.so.2 => /lib/x86_64-linux-gnu/libtbb.so.2 (0x00007fc4e97e0000)
        [...]
	libfreeimage.so.3 => /lib/x86_64-linux-gnu/libfreeimage.so.3 (0x00007fc4e61bf000)
	libopenal.so.1 => /lib/x86_64-linux-gnu/libopenal.so.1 (0x00007fc4e60d7000)
	libcef.so => not found
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc4e60bc000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc4e5eca000)
        [...]
```
Because `libcef.so` is 'not found' it's subdependencies in turn remain invisible for ldd and do not end up in the copy list.

If you change line 36 to `/source/shell/copy_deps.sh /staging/bin/casparcg /staging/lib` `libcef.so` is found and all dependencies end up in the final image. 

Is there a reason why the `ldd` check is done on `/build/shell/casparcg` instead of `/staging/bin/casparcg`?

Should line 36 be changed to `/source/shell/copy_deps.sh /staging/bin/casparcg /staging/lib` or is there a better way to make sure `libcef.so` is found during the `copy_deps` process?

saltomodules commented:
Ok, i made following adjustments:

* Reverted the installation of the missing runtime dependencies in the final docker image. The installation of the `tzdata` package is still there. This makes it possible to set a timezone, with the `TZ` environment variable, when running the container so timestamps in the CasparCG logs reflect the correct time.
* Restored the `LD_LIBRARY_PATH` flag in run.sh. `RPATH ` flags for CMake are still present.
* Changed the `copy_deps.sh` path to ` /staging/bin/casparcg `.

Made a test build from the last commit in this pull request and that seems to result in a fully functional image with all the missing dependencies present.
